### PR TITLE
fix(Load preprocessor symbols): Correctly load preprocessor symbols so they can be used

### LINKED ISSUE #1828: Declare standard predefined symbols when building mutated assemblies
**Is your feature request related to a problem? Please describe.**
See issue #1824 

**Describe the solution you'd like**
Stryker needs to declare predefined symbols when building the mutated assembly.

**Describe alternatives you've considered**
Those symbols can be explicitly declared in the (VS/VC) project settings, but this defeats the purpose of predefined symbols.

**Additional context**
It appears Roslyn does not add them automatically. So the code may have to declare them according the target framework version. Any official method would obviously be preferable.

### PR DESCRIPTION
**Pull request by richardwerkman**
Also add nice hint for opening the report
closes #1828

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core/Reporters/DashboardReporter.cs [hunk 1]
```diff
@@ -39,13 +39,17 @@ public void OnAllMutantsTested(IReadOnlyProjectComponent reportComponent)
...
-                _logger.LogDebug("Your stryker report has been uploaded to: \n {0} \nYou can open it in your browser of choice.", reportUri);
-                _consoleWriter.Write(Output.Green($"Your stryker report has been uploaded to: \n {reportUri} \nYou can open it in your browser of choice."));
-
                 if (_options.ReportTypeToOpen == Options.Inputs.ReportType.Dashboard)
                 {
                     _processWrapper.Open(reportUri);
                 }
+                else
+                {
+                    _consoleWriter.Write(Output.Cyan("Hint: by passing \"--open dashboard or -o dashboard\" the report will open automatically once stryker is done."));
```
Comment (psfinaki) [inline src/Stryker.Core/Stryker.Core/Reporters/DashboardReporter.cs:48]: Is this correct? Per [this](https://github.com/stryker-mutator/stryker-net/pull/1795/files#diff-17ed18489a956f326ec0fe4040850c5bc9261d4631fb42da4c52891d74a59180) it should be something like `--open-report:dashboard`.

File: src/Stryker.Core/Stryker.Core/Reporters/HtmlReporter/HtmlReporter.cs [hunk 2]
```diff
@@ -43,14 +43,17 @@ public void OnAllMutantsTested(IReadOnlyProjectComponent reportComponent)
...
-
             if (_options.ReportTypeToOpen == Options.Inputs.ReportType.Html)
             {
                 _processWrapper.Open(reportUri);
+            } else
+            {
+                _consoleWriter.Write(Output.Cyan("Hint: by passing \"--open or -o\" the report will open automatically once stryker is done."));
             }
+
+            _consoleWriter.WriteLine(Output.Green($"\nYour html report has been generated at: \n " +
```
Comment (psfinaki) [inline src/Stryker.Core/Stryker.Core/Reporters/HtmlReporter/HtmlReporter.cs:54]: Do you need spaces before and after the break?

File: src/Stryker.Core/Stryker.Core/Reporters/DashboardReporter.cs (hunk 1 above)
Comment (psfinaki) [reply src/Stryker.Core/Stryker.Core/Reporters/DashboardReporter.cs:48]:
```suggestion
                    _consoleWriter.Write(Output.Cyan("Hint: by passing \"--open dashboard or -o dashboard\" the report will open automatically once Stryker is done."));
```

**Review by psfinaki — APPROVED** [review]
Put a few small notes BUT the important part is that symbols are now working correctly, manually double-checked this against a repro from the branch :)

If compiled against .NET 6:
image

If compiled against .NET 5:
image

So just what we need and no more weird behavior. Great job @richardwerkman! We'll benefit from that in 6 projects so would be awesome to see this released ASAP :)

File: src/Stryker.Core/Stryker.Core/Reporters/DashboardReporter.cs (hunk 1 above)
Comment (richardwerkman) [reply src/Stryker.Core/Stryker.Core/Reporters/DashboardReporter.cs]: My bad 🙈 apparently I don't know Stryker as well as you do haha

**Comment by richardwerkman** [comment]
@psfinaki v1.2.1 has been released!

**Comment by psfinaki** [comment]
Thanks for the release, will try this out today :)

As for the new diff, sure it's very slick! I mean it's not that I was extremely annoyed by the previous version so I just take this as a facelift. I used to use that "show all" thing occasionally, but it wasn't that important. So yeah good stuff!

**Pull request merged**